src/order/lattice.lean

```diff
@@ -174,6 +177,12 @@ sup_eq_right.2 h
 @[simp] theorem right_eq_sup : b = a ⊔ b ↔ a ≤ b :=
 eq_comm.trans sup_eq_right
 
+@[simp] theorem right_lt_sup : b < a ⊔ b ↔ ¬a ≤ b :=
+le_sup_right.lt_iff_ne.trans $ not_congr right_eq_sup
+
+lemma left_or_right_lt_sup (h : a ≠ b) : (a < a ⊔ b ∨ b < a ⊔ b) :=
```

Comment:
Is this true and/or worth having as an iff?

Reply:
We have `simp` lemmas for `a < a ⊔ b` and `b < a ⊔ b`.